great_expectations/datasource/fluent/spark_filesystem_datasource.pyi

```diff
@@ -41,20 +47,206 @@ class SparkFilesystemDatasource(_SparkFilePathDatasource):
         batch_metadata: Optional[BatchMetadata] = ...,
         batching_regex: re.Pattern | str = r".*",
         glob_directive: str = "**/*",
-        header: bool = ...,
-        infer_schema: bool = ...,
         order_by: Optional[SortersDefinition] = ...,
+        # vvv spark parameters for pyspark.sql.DataFrameReader.csv() (ordered as in pyspark v3.4.0)
+        # path: PathOrPaths,
+        # NA - path determined by asset
+        # schema: Optional[Union[StructType, str]] = None,
+        spark_schema: Optional[Union[pyspark_types.StructType, str]] = None,
+        # sep: Optional[str] = None,
+        sep: Optional[str] = None,
+        # encoding: Optional[str] = None,
+        encoding: Optional[str] = None,
+        # quote: Optional[str] = None,
+        quote: Optional[str] = None,
+        # escape: Optional[str] = None,
+        escape: Optional[str] = None,
+        # comment: Optional[str] = None,
+        comment: Optional[str] = None,
+        # header: Optional[Union[bool, str]] = None,
```

Comment:
@anthonyburdi Curious -- what is the reason for adding the same line with a comment?  Is this something our documentation system looks for?  Thanks.

Reply:
This is a good question. I copied over the signature from the pyspark source code to check whether what I was adding was consistent. Now that they are exactly the same I think I can remove these and instead keep the comments that this set of params is for spark 3.4.0.

Reply:
Actually I think for now I would like to leave them in since they are not exactly the same in many cases (e.g. spark_schema and the spark params are in camel case). It shows the correspondence between spark and our parameter names. This is also in the asset model as aliases, so I may be convinced at some point to remove these but I think they will be helpful when we add support for other spark versions.